services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -484,26 +512,36 @@ async def get_project_access_rights_for_user(
         """
         User project access rights. Aggregated across all his groups.
         """
-        # NOTE: MD: I didn't manage to write this query in sqlalchemy, but
-        # when we migrate project access rights to its own table, this will
-        # be not needed and we can use something similar to "list_wallets_for_user"
-        raw_sql = sa.text(
-            f"""
-                select uid, max(read) as read, max(write) as write, max(delete) as delete
-                from user_to_groups utg
-                join
-                    (
-                        SELECT cast(key as int) AS id, value ->>'read' as read, value ->>'write' as write, value ->>'delete' as delete
-                        FROM projects, jsonb_each(access_rights)
-                        where "uuid" = '{project_uuid}'
-                    ) prj_access_rights on utg.gid = prj_access_rights.id
-                where utg.uid = {user_id}
-                group by uid
-            """  # noqa: S608
+        _SELECTION_ARGS = (
+            user_to_groups.c.uid,
+            func.max(project_to_groups.c.read.cast(INTEGER))
+            .cast(BOOLEAN)
+            .label("read"),
+            func.max(project_to_groups.c.write.cast(INTEGER))
+            .cast(BOOLEAN)
+            .label("write"),
+            func.max(project_to_groups.c.delete.cast(INTEGER))
+            .cast(BOOLEAN)
+            .label("delete"),
+        )
+
+        _JOIN_TABLES = user_to_groups.join(
+            project_to_groups, user_to_groups.c.gid == project_to_groups.c.gid
+        )
+
+        stmt = (
+            sa.select(*_SELECTION_ARGS)
+            .select_from(_JOIN_TABLES)
+            .where(
+                (user_to_groups.c.uid == user_id)
+                & (project_to_groups.c.project_uuid == f"{project_uuid}")
+                & (project_to_groups.c.read == "true")
```

Comment:
Q: @pcrespov @GitHK @odeimaiz  What do you think is valid that the user needs at least read access in the `project_to_groups` table to see the project? (This was ignored till now, but it seems reasonable to introduce this requirement, correct?)

Reply:
Currently I see nothing against it but maybe @sanderegg or @pcrespov have other opinions.

Reply:
yes, it is a valid argument. thx